### PR TITLE
importinto/lightning: fix incorrect datatime value for zero date

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/50757

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/realtikvtest/importintotest/import_into_test.go [hunk 1]
```diff
@@ -1166,3 +1166,20 @@ func (s *mockGCSSuite) TestAnalyze() {
 	}, 60*time.Second, time.Second)
 	s.tk.MustQuery("SHOW ANALYZE STATUS;").CheckContain("analyze_table")
 }
+
+func (s *mockGCSSuite) TestZeroDateTime() {
+	s.tk.MustExec("DROP DATABASE IF EXISTS import_into;")
+	s.tk.MustExec("CREATE DATABASE import_into;")
+
+	s.tk.MustExec("create table import_into.zero_time_table(t datetime)")
+	s.tk.MustExec("set global sql_mode='STRICT_TRANS_TABLES'")
```
Comment (lance6716) [inline tests/realtikvtest/importintotest/import_into_test.go:1175]:
```suggestion
	s.tk.MustExec("set @@sql_mode='STRICT_TRANS_TABLES'")
```

more reasonable to set it for this session

**Review by lance6716 — APPROVED** [review]

File: tests/realtikvtest/importintotest/import_into_test.go (hunk 1 above)
Comment (D3Hunter) [reply tests/realtikvtest/importintotest/import_into_test.go:1175]: need reset it after this test, as `tk` is shared

File: tests/realtikvtest/importintotest/import_into_test.go [hunk 2]
```diff
@@ -1166,3 +1166,20 @@ func (s *mockGCSSuite) TestAnalyze() {
...
+
+	s.server.CreateObject(fakestorage.Object{
+		ObjectAttrs: fakestorage.ObjectAttrs{BucketName: "test-load", Name: "zero_time.csv"},
+		Content:     []byte("1990-01-00 00:00:00\n"),
+	})
+
+	sql := fmt.Sprintf(`IMPORT INTO import_into.zero_time_table FROM 'gs://test-load/zero_time.csv?endpoint=%s'`, gcsEndpoint)
```
Comment (D3Hunter) [inline tests/realtikvtest/importintotest/import_into_test.go:1182]: add a negative case?

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: tests/realtikvtest/importintotest/import_into_test.go (hunk 2 above)
Comment (GMHDBJD) [reply tests/realtikvtest/importintotest/import_into_test.go:1182]: [7fbadd3](https://github.com/pingcap/tidb/pull/51201/commits/7fbadd349ad286a56ff422204893d89e8c0ac39d)

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**